rclpy_components/rclpy_components/component_manager.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from rclpy.node import Node
+from rclpy.executors import Executor
+from rclpy.logging import get_logger
+from rclpy.exceptions import InvalidNodeNameException, InvalidNamespaceException
+from composition_interfaces.srv import ListNodes, LoadNode, UnloadNode
+try:
+    from importlib.metadata import entry_points
+except ImportError:
+    from importlib_metadata import entry_points
+
+RCLPY_COMPONENTS = 'rclpy_components'
+logger = get_logger('ComponentManager')
+
+
+class ComponentManager(Node):
+
+    def __init__(self, executor: Executor, name="py_component_manager", **kwargs):
+        # TODO Handle the py args equivalent to rclcpp 'NodeOptions'
+        super().__init__(name, **kwargs)
+        self.executor = executor
+        # Implement the 3 services described in
+        # http://design.ros2.org/articles/roslaunch.html#command-line-arguments
+        self.list_node_srv_ = self.create_service(ListNodes, "~/_container/list_nodes", self.on_list_node)
```

Comment:
nitpick: The convention for "private" members in Python is to use a leading underscore (not a trailing underscore)
```suggestion
        self._list_node_srv = self.create_service(ListNodes, "~/_container/list_nodes", self.on_list_node)
```

Apply the same naming for other "private" members.